channel/consensus_channel/consensus_channel_test.go

```diff
@@ -109,6 +109,54 @@ func TestConsensusChannel(t *testing.T) {
 		}
 	}
 
+	testApplyingRemoveProposalToVars := func(t *testing.T) {
+		startingTurnNum := uint64(9)
+
+		vars := Vars{TurnNum: startingTurnNum, Outcome: outcome()}
+		aAmount, bAmount := uint64(2), uint64(3)
+		proposal := remove(10, existingChannel, aAmount, bAmount)
+		err := vars.Remove(proposal)
+
+		if err != nil {
+			t.Fatalf("unable to compute next state: %v", err)
+		}
+
+		if vars.TurnNum != startingTurnNum+1 {
+			t.Fatalf("incorrect state calculation: %v", err)
+		}
+
+		expected := makeOutcome(
+			allocation(alice, aBal+aAmount),
+			allocation(bob, bBal+bAmount),
+		)
+
+		if diff := cmp.Diff(vars.Outcome, expected, cmp.AllowUnexported(expected, Balance{}, big.Int{}, Guarantee{})); diff != "" {
+			t.Fatalf("incorrect outcome: %v", diff)
+		}
+
+		// Proposing the same change again should fail since the guarantee has been removed
+		duplicateProposal := proposal
+		duplicateProposal.turnNum += 1
+		err = vars.Remove(duplicateProposal)
+
+		if !errors.Is(err, ErrGuaranteeNotFound) {
+			t.Fatalf("expected error when adding duplicate guarantee: %v", err)
+		}
+
+		// Proposing a remove that cannot be afforded by the guarantee should fail
```

Comment:
What should happen if the `Remove` can be afforded, but doesn't fully use up the guarantee's amount?

Reply:
Good question! I think for the initial implementation it should error. `Remove` is a proposal to remove a guarantee so all the funds in the guarantee should be accounted for.

Reply:
I've tried to make this more explicit in [d78f22e](https://github.com/statechannels/go-nitro/pull/483/commits/d78f22e23567cf6f2e5578f758ff73fd6b53e9c9)

Reply:
👍